osgar/lib/test_config.py

```diff
@@ -13,8 +13,34 @@ def test_empty_config(self):
         conf = Config()
         self.assertIsNone(conf.data.get('localization'))
 
-if __name__ == "__main__":
-    unittest.main() 
+    def test_load_config_files(self):
+        conf_dir = '../../config'
+        filename = test_data('ro2018-spider-gps-imu.json', conf_dir)
+        conf = Config.load(filename)
 
-# vim: expandtab sw=4 ts=4 
+    def test_multiple_config_files(self):
+        conf_dir = '../../config'
+        filename1 = test_data('ro2018-spider-gps-imu.json', conf_dir)
+        filename2 = test_data('ro2018-czu-waypoints.json', conf_dir)
+        conf = Config.load([filename1, filename2]).data
```

Comment:
Here you get only the contents of `filename2`. Or maybe not. Depending if the class `data` attribute is or is not shadowed with an instance one. That is a maintenance nightmare.

Reply:
class `data` is not used - why do you think so? It should be all instances

Reply:
Yes, except https://github.com/robotika/osgar/blob/2cfb1f39f517100fe8635eff9228e762e442353c/osgar/lib/config.py#L35

So you are just shadowing class `data` with instance `data`. As I said in other comment, there is no need for a class. A  function would do just fine.